### ISSUE TITLE
Use x360ce with two ps4 controllers on pass trough

### ISSUE THREAD

**Issue by memphisobrien** [x360ce/x360ce#435]
Hi, I use DS4 for my 2 ps4 controllers to use it as 360 controllers with 360ce. 

Sadly 2 LT + RT together earse eachother (cant aim and shoot at the same time)

Now I found the solution to just mark pass through in the options. Sadly it only works with 1 controller (here aim and shoot works now). But the second controller with pass trough just converts with pass trough just in a copy of a first (a button on the first ps4 controller shows in controller 1 with pass trough and the second controller with pass trogh (but I mapped both to different numbers)).

How to use 2 ps4 controllers in 360ce with pass trough?

Thank you a lot

**Comment by Squall-Leonhart** [comment]
You don't?

passthrough is for common class game controllers.

**Comment by R3cK17** [comment]
can you recommend anything purely for adjusting deadzones on thumbsticks? I actually tried Durazno before this, all it would do was crash GTAV on startup using the x64 wrapper 1_3.dll and i have the notorious thumbstick that does not spring back to a tolerated deadzone :-/

**Comment by Melek68** [comment]
Have the same problem with 2 Xbox 360 controllers. Is there any possibility that the triggers on both controllers work correctly? I know they are both xinput, anyway I need this program for ASTER.
PLEASE HELP US!!
Edit: Durazno/KrossX wouldn't help me

**Comment by Nucleoprotein** [comment]
Use passtrough, if you want only to reconfigure buttons for XBOX controller x360ce is NOT for you use KrossX Durazno: https://github.com/KrossX/Durazno

**Comment by Melek68** [comment]
yeah I tried that, but it only works for the first controller like memphisobrien said:

> Now I found the solution to just mark pass through in the options. Sadly it only works with 1 controller (here aim and shoot works now). But the second controller with pass trough just converts with pass trough just in a copy of a first (a button on the first ps4 controller shows in controller 1 with pass trough and the second controller with pass trogh (but I mapped both to different numbers)).

**Comment by Nucleoprotein** [comment]
Passtrough works for any controller, select proper "Pass Trough ID" for each controller.

**Comment by Melek68** [comment]
Sorry but I also tried that. When I change the pass through ID for each controller it makes no difference. First controller works, second controller doesn't (it uses the first controller instead). I don't know... Have you got another solution or should I give up?

**Comment by Nucleoprotein** [comment]
Selecting "Any" for both also not fix problem ?

**Comment by Melek68** [comment]
nope
Edit: only to give more information: When I disable pass through, the second controller works but only with the trigger problem... When I enable Pass through the first controller reacts instead of the second.

**Comment by R3cK17** [comment]
Damn I didn't mean to make this blow up :/

Try "Xinput Plus" it works perfect for me and has a heap of cool options. I was able to offset the center and increase the deadzones to fix my thumbstick drift.

**Comment by Melek68** [comment]
Pleaaase isn't there a possibility in !!x360ce!! to enable pass through for both xinput-controllers that both will work? I'm despairing....

**Comment by Nucleoprotein** [comment]
I think last change in passtrough code was done long ago by @jbienz and nobody found any bugs then ...

**Comment by Squall-Leonhart** [comment]
Err... actually, this was raised on the forum some time ago.

**Comment by Nucleoprotein** [comment]
https://github.com/x360ce/x360ce/blob/master/x360ce/x360ce/Config.cpp#L330 :expressionless: 
Setting is ignored in full passtrough, so always index 0 is used ...

Edit: try this DLL (x86 only): http://www68.zippyshare.com/v/y4RzMTvY/file.html

**Comment by Nucleoprotein** [comment]
Pushed small fix to master that add possibility to set `PassThroughIndex` to `-1` then is used default user index.

**Comment by Melek68** [comment]
Thank you for your effort, but this also didn't work for me. I "fixed" it by disabling pass through on the second controller and changed the places of LB and LT. Then I re-changed the controls ingame --> the gamepad works without the trigger-issue :)
I think it's not the perfect solution but it works...

**Comment by Nucleoprotein** [comment]
@Squall-Leonhart 
Can you check ?

**Comment by Squall-Leonhart** [comment]
unfortunately can't. i only have the one controller and i don't even use the xusb driver for it.

**Comment by ghost** [comment]
I just wanted to bump this to say that I have the same problem (although its for MK X for me), any progress ?

**Comment by EvoJocys** [comment]
This report is about the 3.x line, where x360ce was an xinput DLL copied into each game's folder. The current 4.x line works differently: it creates a virtual Xbox 360 controller that every game sees, with nothing copied into game folders. Game-specific problems from the DLL era do not carry over.

The latest release is 4.21 (September 2026): https://github.com/x360ce/x360ce/releases

Closing as part of a tracker clean-up after the 4.21 release. If it still happens on 4.21, reopen this issue with the version and what you see.
